STM32F405RGTx/Core/Inc/state_machine.hpp

```diff
@@ -0,0 +1,104 @@
+/* state_machine.h
+ *
+ *    Created on: Jul. 11, 2021
+ *            Author: Kanishk Kashyap
+ */
+#pragma once
+
+#include "can.h"
+#include "cmsis_os.h"
+#include "can.h"
+#include "util.hpp"
+
+#ifndef _STATE_MACHINE_H_
+#define _STATE_MACHINE_H_
+
+// Constants - These numbers are arbritrary should be changed after testing
+// Mosfet Array Length
+#define NUM_MOSFETS 6
+#define NUM_PHASE_OUTPUTS 3
+
+// Severe Fault
+#define MAX_MOSFET_TEMP_SEVERE 70.0
+#define MAX_DCVOLTAGE_SEVERE 4.0
+#define MIN_DCVOLTAGE_SEVERE 1.8
+#define MAX_VOLTAGE_SEVERE 4.0
+#define MIN_VOLTAGE_SEVERE 1.0
+#define MAX_TEMP_SEVERE 70.0
+#define MAX_CURRENT_SEVERE 50.0
```

Comment:
We should talk more with the electrical team about these thresholds. If you copied them over from BMS, they're likely outdated. IK off the top of my head that the new MC board should be (theoretically) able to do 60A without any issues.

Reply:
Yea these values are completely dummy, I already let Dev know and he said he'll replace them once they have more information.